ui-service/src/main/frontend/hotel-booking-ui/src/components/search-components/RoomSelectorComponent.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { User, Minus, Plus } from 'lucide-react';
+import './styles/RoomSelector.css';
+
+interface SearchHotelComponentProps {
+    rooms: number;
+    adults: number;
+    children: number;
+    onRoomDataChange: (key: 'rooms' | 'adults' | 'children', value: number) => void;
+}
+
+const SearchHotelComponent: React.FC<SearchHotelComponentProps> = ({ rooms, adults, children, onRoomDataChange }) => {
+
+    // selector section
+    const [isOpen, setIsOpen] = useState(false);
+    const dropdownRef = useRef<HTMLDivElement>(null);
+    // Close dropdown when clicking outside the component
+    useEffect(() => {
+        const handleClickOutside = (event: MouseEvent) => {
+            if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+                setIsOpen(false);
+            }
+        };
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, []);
+
+    // Handle increment/decrement for rooms, adults, and children
+    const handleChange = (key: 'rooms' | 'adults' | 'children', increment: boolean) => {
+        const currentValue = key === 'rooms' ? rooms : key === 'adults' ? adults : children;
+        const newValue = increment ? currentValue + 1 : currentValue - 1;
+        onRoomDataChange(key, newValue);
+    };
+
+    // Render counter controls for rooms, adults, and children
+    const renderCounter = (label: string, key: 'rooms' | 'adults' | 'children') => (
```

Comment:
If the child counter is greater than 1, it will show a row to prompt child age for each of the children